x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -313,7 +313,7 @@ private <T> void preAnalyze(
             // First resolve the lookup indices, then the main indices
             preAnalyzeLookupIndices(
                 preAnalysis.lookupIndices,
-                fieldNames,
+                Set.of("*"), // Current LOOKUP JOIN syntax does not allow for field selection
```

Comment:
Temporary fix to work around a bug in fieldNames for lookup-join when there are aliases before, and keep after the lookup.